inference/wasm/tests/engine/translations-engine.worker.mjs

```diff
@@ -0,0 +1,432 @@
+
+    for (const [fileType, file] of Object.entries(
+      languageTranslationModelFiles,
+    )) {
+      const alignment = MODEL_FILE_ALIGNMENTS[fileType];
+      if (!alignment) {
+        throw new Error(`Unknown file type: "${fileType}"`);
+      }
+
+      /** @type {AlignedMemory} */
+      const alignedMemory = new bergamot.AlignedMemory(
+        file.buffer.byteLength,
+        alignment,
+      );
+      alignedMemory.getByteArrayView().set(new Uint8Array(file.buffer));
+
+      results[fileType] = alignedMemory;
+    }
+
+    return results;
+  }
+
+  /**
+   * Initializes the Bergamot WASM module.
+   *
+   * @param {ArrayBuffer} wasmBinary - The WASM binary data.
+   * @returns {Promise<BergamotModule>} Resolves with the initialized Bergamot module.
+   */
+  static initializeWasm(wasmBinary) {
+    return new Promise((resolve, reject) => {
+      /** @type {BergamotModule} */
+      const bergamot = loadBergamot({
+        INITIAL_MEMORY: 234_291_200,
+        print: () => {},
+        // Uncomment this line to display logs in tests.
+        // To show logs, run with the --runner=basic flag.
+        // print: (...args) => console.log(...args),
+        onAbort() {
+          reject(new Error("Error loading Bergamot WASM module."));
+        },
+        onRuntimeInitialized: () => {
+          try {
+            resolve(bergamot);
+          } catch (e) {
+            reject(e);
+          }
+        },
+        wasmBinary,
+      });
+    });
+  }
+
+  /**
+   * Generates a configuration string for the Marian translation service.
+   *
+   * @param {Record<string, string>} config - Configuration key-value pairs.
+   * @returns {string} Formatted configuration string.
+   */
+  static generateTextConfig(config) {
+    const indent = "            ";
+    let result = "\n";
+
+    for (const [key, value] of Object.entries(config)) {
+      result += `${indent}${key}: ${value}\n`;
+    }
+
+    return result + indent;
+  }
+
+  /**
+   * Prepares translation arguments for the Bergamot module.
+   *
+   * @param {BergamotModule} bergamot - Initialized Bergamot module.
+   * @param {string} sourceText - Text to translate.
+   * @param {boolean} isHTML - Indicates if the text contains HTML.
+   * @returns {{messages: VectorString, options: VectorResponseOptions}} Prepared messages and options.
+   */
+  static getTranslationArgs(bergamot, sourceText, isHTML) {
+    /** @type {VectorString} */
+    const messages = new bergamot.VectorString();
+    /** @type {VectorResponseOptions} */
+    const options = new bergamot.VectorResponseOptions();
+
+    if (sourceText) {
+      messages.push_back(sourceText);
+      options.push_back({
+        qualityScores: false,
+        alignment: true,
+        html: isHTML,
+      });
+    }
+
+    return { messages, options };
+  }
+}
+
+/**
+ * Regular expression to match whitespace before and after the text.
+ *
+ * @type {RegExp}
+ */
+const whitespaceRegex = /^(\s*)(.*?)(\s*)$/s;
```

Comment:
Thought: Hmm... this is a bit of a maintenance burden since we now have code duplication. A good example of this is in the whitespaceRegex code. It'll be a bit hard to keep the two implementations up to date with each other. I'm not sure what our long term plan here with this code. We should probably discuss it a bit (not in this PR). I know we had some conversation on Matrix regarding perf testing code, which is basically this same problem.

Reply:
I'm definitely on the same page in general. I mentally justified doing this because this implementation is an extremely simplified/streamlined version of what is in the `mozilla-unified` tree. That is, it doesn't deal with work queues with async time slices. It doesn't manage multiple translators with keep-alive timeouts etc.

The translations engine here just has room to translate one single message at a time, and that's all it does. I actually think there is a bit of value in having a bare-bones example of the connections without the Firefox complexity, even if there is a bit of code duplication. 

Regarding the whitespaceRegex, I almost just excluded it entirely, and I might still take it out, so I'm going to send this comment back your way for a response. We don't really _need_ to handle the whitespace here in our test cases. 

So, a few follow-up questions:

1) Do you similarly see value in having this bare-bones implementation even if some code is duplicated?
2) Do you agree that we can remove the whitespaceRegex here in the tests, as that's more of a Firefox-algorithm data cleaning detail?
3) Where would you like to have this broader conversation about code duplication?

Reply:
1. Yes I think it's fine for now, more of a long term thinking.
2. Yeah I think it can be removed, but doesn't hurt to land initially.
3. I would say let me do some of my perf work in here to get a feel for the repo and how it feels to work with it, then we can follow up in Matrix or a team meeting to discuss. If we need to we can write a document, but I don't think it needs to be that formal.